caffe2/operators/hardtanh_op.cu

```diff
@@ -0,0 +1,69 @@
+#include "caffe2/core/common_gpu.h"
+#include "caffe2/core/context_gpu.h"
+#include "caffe2/operators/hardtanh_op.h"
+
+namespace caffe2 {
+namespace {
+template <typename T>
+__global__ void HardtanhKernel(const int N, const T* X, T* Y, T min_val_, T max_val_) {
+  // Utilize naive implementation of Hardtanh
+  CUDA_1D_KERNEL_LOOP(i, N) {
+    Y[i] = X[i] > max_val_ ? max_val_ : (X[i] < min_val_ ? min_val_ : X[i]);
```

Comment:
I'd suggest to use fmaxf and fminf instread of conditions in cuda kernels.

Reply:
Would you recommend this for the slightly more complex HardtanhGradientKernel, as well?